tests/AutoReview/ProjectCodeTest.php

```diff
@@ -763,12 +818,15 @@ public function testConstantsAreInUpperCase(string $className): void
 
     private function createTokensForClass(string $className): Tokens
     {
-        $file = $className;
-        $file = preg_replace('#^PhpCsFixer\\\Tests\\\#', 'tests\\', $file);
-        $file = preg_replace('#^PhpCsFixer\\\#', 'src\\', $file);
-        $file = str_replace('\\', \DIRECTORY_SEPARATOR, $file).'.php';
+        $filename = (new \ReflectionClass($className))->getFileName();
+
+        static $fileTokens = [];
```

Comment:
I like the cache. Let's clear it in `tearDownAfterClass` ?

Reply:
For this single cache variable only? If we want to do it let's do it at least in all `AutoReview` tests (in separate PR and with measuring if it has an impact on performance).

Reply:
I wonder when the garbage collector clear it, if any. back in time PHPUnit was _not_ unregistering instances of test classes, eg `ProjectCodeTest`, and `static` variable remained defined forever, keeping the place in memory.

Reply:
@kubawerlos ,

`static $foo` within a method works exactly same way as within a function. It creates global object. PHP will have no clue when to free memory of this global object, because it doesn't know if it will be used again later.

And it behave same way as in function, meaning that it's state is shared within all instances - including future instances. Thus even if TestCase class instance would get `unset()`, PHP will still keep static(==global) variable in memory, in case any other instance would be created and method called, eg https://3v4l.org/3iRb0

moving this to static property of class allows us to instruct PHP to mark this property as free-able by GC, by putting `= null` instruction in `tearDownAfterClass`.

---

and here, you create , in that global variable, over 1_000 tokens collections